utilities/transactions/pessimistic_transaction_db.cc

```diff
@@ -681,5 +720,20 @@ bool WritePreparedTxnDB::ExchangeCommitEntry(uint64_t indexed_seq,
   return true;
 }
 
+bool WritePreparedTxnDB::MaybeUpdateOldCommitMap(const uint64_t& prep_seq,
+                                                 const uint64_t& commit_seq,
+                                                 const uint64_t& snapshot_seq) {
+  if (commit_seq <= snapshot_seq) {
```

Comment:
If we do not store an entry in old_commit_map we assume it is committed in all snapshots. if commit_seq == snapshot_seq, it is considered already in the snapshot so we need not to keep the entry around.